src/control/cmd/dmg/system.go

```diff
@@ -34,11 +34,31 @@ import (
 
 // SystemCmd is the struct representing the top-level system subcommand.
 type SystemCmd struct {
+	LeaderQuery leaderQueryCmd       `command:"leader-query" alias:"l" description:"Query for current Management Service leader"`
 	MemberQuery systemMemberQueryCmd `command:"member-query" alias:"q" description:"Retrieve DAOS system membership"`
 	Stop        systemStopCmd        `command:"stop" alias:"s" description:"Perform controlled shutdown of DAOS system"`
 	ListPools   systemListPoolsCmd   `command:"list-pools" alias:"p" description:"List all pools in the DAOS system"`
 }
 
+type leaderQueryCmd struct {
+	logCmd
+	cfgCmd
+	connectedCmd
+}
+
+func (cmd *leaderQueryCmd) Execute(_ []string) error {
```

Comment:
Would there be value in making a command for current leader and a separate query command for replica set?  I'm not clear why we want to view the leader and the replica at the same time if the query is really a 'leader' query.  I am not saying it is a problem.  I'm just thinking out loud if we'd be better off splitting it up.

Reply:
This work is mostly about putting the plumbing in place for advanced Management Service clients to determine the current leader rather than using a hard-coded list. Since we're not going to support more than one MS replica in the 1.0 timeframe, I decided not to complicate that code until we need to. In the meantime, I thought that having the functionality available via a dmg command could be useful for an admin who wants to know what the configured access points are without going to look at the server config. It doesn't make sense to me to split them up -- as an admin I think I'd be annoyed to have to run two commands to find the closely-related information.

Reply:
That makes sense and is good.  Thanks for the explanation of what you had in mind.

Reply:
Incidentally, the replica set will be returned as part of the query because it's possible for there to be a leadership change in between querying for the current leader and making a request. The thought when we discussed this at the f2f was that we'd return the current leader as a best guess and if that doesn't work out the client will try the other replicas until it succeeds.